File: backend/schemas/housing.py
from datetime import date, datetime
from decimal import Decimal
from typing import List

from pydantic import BaseModel, Field


class HousingBase(BaseModel):
    title: str = Field(..., min_length=3, max_length=150)
    description: str
    rent: Decimal
    deposit: Decimal
    city: str
    locality: str
    address: str
    house_type: str
    sharing_type: str
    gender_preference: str
    is_furnished: bool
    available_from: date
    contact_number: str
    images: List[str]


class HousingCreate(HousingBase):
    pass


class HousingUpdate(HousingBase):
    pass


class HousingResponse(HousingBase):
    id: int
    owner_id: int
    available: bool
    verified: bool
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }


class HousingListResponse(BaseModel):
    items: List[HousingResponse]
    total: int
    total_pages: int

File: src/types/housing.ts
export interface Housing {
  id: string;

  owner_id: string;

  created_at: string;

  updated_at: string;

  title: string;

  description: string;

  rent: number;

  deposit: number;

  city: string;

  locality: string;

  address: string;

  latitude?: number;

  longitude?: number;

  house_type: string;

  sharing_type: string;

  gender_preference: string;

  is_furnished: boolean;

  available_from: string;

  available: boolean;

  verified: boolean;

  contact_number: string;

  images: string[];
}



export interface HousingFilters {

  city?: string;

  locality?: string;

  max_rent?: number;

  house_type?: string;

  sharing_type?: string;

  gender_preference?: string;

  available?: boolean;

  is_furnished?: boolean;

}



export interface CreateHousingRequest {

  title: string;

  description: string;

  rent: number;

  deposit: number;

  city: string;

  locality: string;

  address: string;

  house_type: string;

  sharing_type: string;

  gender_preference: string;

  is_furnished: boolean;

  available_from: string;

  contact_number: string;

  images: string[];

}



export interface UpdateHousingRequest {

  title?: string;

  description?: string;

  rent?: number;

  deposit?: number;

  city?: string;

  locality?: string;

  address?: string;

  house_type?: string;

  sharing_type?: string;

  gender_preference?: string;

  is_furnished?: boolean;

  available_from?: string;

  contact_number?: string;

  images?: string[];

}



export interface HousingResponse {

  data: Housing;

}



export interface HousingListResponse {

  items: Housing[];

  total?: number;

  total_pages?: number;

}

export type HousingListing = Housing;